addOns/scripts/src/main/java/org/zaproxy/zap/extension/scripts/automation/ExtensionScriptAutomation.java

```diff
@@ -103,4 +130,80 @@ public static String getResourceAsString(String name) {
         }
         return "";
     }
+
+    private class ScriptErrorHandler implements ScriptEventListener, EventConsumer {
+
+        private List<AutomationPlan> runningPlans = Collections.synchronizedList(new ArrayList<>());
+
+        @Override
+        public void refreshScript(ScriptWrapper script) {
+            // Nothing to do.
+        }
+
+        @Override
+        public void scriptAdded(ScriptWrapper script, boolean display) {
+            // Nothing to do.
+        }
+
+        @Override
+        public void scriptRemoved(ScriptWrapper script) {
+            // Nothing to do.
+        }
+
+        @Override
+        public void preInvoke(ScriptWrapper script) {
+            // Nothing to do.
+        }
+
+        @Override
+        public void scriptChanged(ScriptWrapper script) {
+            // Nothing to do.
+        }
+
+        @Override
+        public void scriptError(ScriptWrapper script) {
+            if (ExtensionScript.TYPE_STANDALONE.equals(script.getTypeName())) {
+                // Errors of stand alone scripts are handled directly by the job.
+                return;
+            }
+
+            synchronized (runningPlans) {
+                runningPlans.forEach(
+                        plan ->
+                                plan.getProgress()
```

Comment:
If I understand correctly, this will add the script errors to all running plans regardless of whether they loaded the specific script or not. Is there any way to check that this plan actually loads the script with the error?

Reply:
We could do that but then we would miss errors of scripts that were loaded/enabled by other means/beforehand (e.g. community scripts).